### DISCUSSION TITLE
CoaxCopter Mixing question

### DISCUSSION THREAD

**Discussion by ansib1e** [iNavFlight/inav#9750] [Q&A]
Hello everyone,

I am trying to build a CoaxCopter with a pair of contra rotating motors on top and 2 servos controlling a gimbal/TVC system for pitch and roll (similar to designs shown in Ardupilot wiki for single/coax copter). I am wondering how to setup the mixer for the motors and servos as this is not a traditional setup. ardupilot has a specific frame type for this and wondering if anyone got a similar system working in iNav?

http://ardupilot.org/copter/docs/singlecopter-and-coaxcopter.html

I am using a Matek F405 VTOL Flight Controller with iNav 7.0

Thanks for everyone's help!

**Comment by sensei-hacker** [comment]
For unusual / custom aircraft like this, it's essentially a matter of doing one output at a time, one axis at a time.

In the mixer, add the first motor.
First motor should of course respond to throttle, so put in throttle.  On coax copters, motors don't do roll and pitch, so put zeros in those two fields. Motors are used for yaw, so put maybe 30% yaw to start.  Should that be positive or negative?  Think about which direction the motor is spinning - the body will spin the opposite direction as that motor is increased. So a clockwise motor gets positive yaw. You can confirm by holding it and feeling the body twist to the right when you give it a bit of throttle.

Make sure you have the first motor responding as you want.

Then add the next motor, same settings just with yaw reversed (negative or positive).

Once both motors are responding the way you want, add the first servo. What do you want that servo to do for roll?  If that servo is for roll, put in 100% roll.  Check the direction. Reverse it if needed.

Once roll is how you want it, should that servo respond to pitch? (Probably not, if you only have two servos).

After the first servo is responding the way you need it, add the second servo.

The way to make things complicated is to try to think about all the axis on all the outputs all at once, and go messing with them all somewhat randomly.  If you avoid that and just do one at a time, it will probably be reasonably straightforward for you.

**Reply by ansib1e** [reply]
Thank you so much for this very thorough explanation! If I understood correctly for my case, I would be using the standard multirotor/quad setup in the mixer, adding the two motors on s1 and s2 (throttle 100% and yaw 30% for both) and the two servos on s9 and s10 (stabilized pitch 100% for s9 and stabilized roll 100% for s10?). This really helped clarify a lot and I'll follow your advice on doing it one axis at a time, I'm just a bit confused still on the actual mixing for specifically the Matek F405 VTOL. It says in the manual for it that motors are s1 - s8 for multirotors but s1 to s2 for planes and not sure which one this craft fits under. Thanks again so much for your help!

**Reply by sensei-hacker** [reply]
Sorry I didn't see your reply earlier. This is probably too late, but:

> It says in the manual for it that motors are s1 - s8 for multirotors but s1 to s2 for planes and not sure which one this craft fits under.

Since you have two motors, they'll probably be on S1 and S2. You probably want to start with multirotor craft type since you don't have wings.

You can click in the mixer to add servos and it will show which output numbers they end up on.

It doesn't look like anyone has come along with that exact setup using that exact FC in order to share their specific configuration. 

Without copying someone else's exactly, it just comes down to:
Get yaw going the right way with one of the motors. 

Get yaw going the right way on the other motor.

Then don't touch the yaw again - it's done.

Get one servo going as you want it for pitch.
...

Just one output at a time, setup one axis at a time, without going back and messing up what you've already done.
